components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1496,6 +1497,8 @@ public Response deleteAPI(String apiId, String ifMatch, MessageContext messageCo
                     if (apiUsages != null && apiUsages.size() > 0) {
                         RestApiUtil.handleConflict("Cannot remove the API " + apiId + " as active subscriptions exist", log);
                     }
+                } catch (ConflictException e) {
+                    throw new APIManagementException("Error occurred while deleting API: " + apiId, e);
```

Comment:
At the bottom of the method also, lets remove the internal server error throwing part and just throw the error.

At the end of the day, the client should get a 409 Conflict response and the API should not be deleted.

```
        } catch (APIManagementException e) {
            //Auth failure occurs when cross tenant accessing APIs. Sends 404, since we don't need to expose the existence of the resource
            if (RestApiUtil.isDueToResourceNotFound(e) || RestApiUtil.isDueToAuthorizationFailure(e)) {
                RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API, apiId, e, log);
            } else if (isAuthorizationFailure(e)) {
                RestApiUtil.handleAuthorizationFailure("Authorization failure while deleting API : " + apiId, e, log);
            } else {
                throw e;
            }
        }
```

Reply:
Catch the ConflictException in separate block and throws